nomos-services/data-availability/network/src/opinion_aggregator.rs

```diff
@@ -28,12 +28,27 @@ pub enum OpinionError {
     Error(#[from] DynError),
 }
 
-pub struct Opinions {
-    pub session_id: SessionNumber,
-    pub new_opinions: BitVec,
-    pub old_opinions: BitVec,
+pub struct ActivityProof {
+    pub current_session: SessionNumber,
+    pub previous_session_opinions: BitVec<u8, Lsb0>,
+    pub current_session_opinions: BitVec<u8, Lsb0>,
 }
 
+impl ActivityProof {
+    #[expect(
+        dead_code,
+        reason = "This will be used as soon as sdp adapter is connected"
+    )]
+    pub fn into_metadata_bytes(self) -> Vec<u8> {
+        let mut bytes = vec![0x01]; // Version byte from spec
+        bytes.extend(&self.current_session.to_le_bytes());
+        let prev_bytes = bitvec_to_bytes(self.previous_session_opinions);
+        bytes.extend(prev_bytes);
+        let curr_bytes = bitvec_to_bytes(self.current_session_opinions);
+        bytes.extend(curr_bytes);
+        bytes
+    }
```

Comment:
Is this up per spec?

Reply:
This might later be changed to an enum in `nomos-core`, I think it'll be more ergonomic for use in the ledger.